test/skimNtuple2018_HHbtag.cpp

```diff
@@ -940,21 +978,48 @@ int main (int argc, char** argv)
       if (njets > 4) njets = 4;
       if (nb < 0)    nb = 0;
       if (nb > 4)    nb = 4;
-
-      float ht = theBigTree.lheHt;
-      int nht = 0;
-      if      (ht < 0                ) nht = 0;
-      else if (ht >= 0   && ht <70   ) nht = 0;
-      else if (ht >= 70   && ht <100 ) nht = 1;
-      else if (ht >= 100  && ht <200 ) nht = 2;
-      else if (ht >= 200  && ht <400 ) nht = 3;
-      else if (ht >= 400  && ht <600 ) nht = 4;
-      else if (ht >= 600  && ht <800 ) nht = 5;
-      else if (ht >= 800  && ht <1200) nht = 6;
-      else if (ht >= 1200 && ht <2500) nht = 7;
-      else /* ht >= 2500*/             nht = 8;
-
-      stitchWeight = stitchWeights[njets][nb][nht];
+      
+      //float ht = theBigTree.lheHt;
+      //int nht = 0;
+      //if      (ht < 0                ) nht = 0;
+      //else if (ht >= 0   && ht <70   ) nht = 0;
+      //else if (ht >= 70   && ht <100 ) nht = 1;
+      //else if (ht >= 100  && ht <200 ) nht = 2;
+      //else if (ht >= 200  && ht <400 ) nht = 3;
+      //else if (ht >= 400  && ht <600 ) nht = 4;
+      //else if (ht >= 600  && ht <800 ) nht = 5;
+      //else if (ht >= 800  && ht <1200) nht = 6;
+      //else if (ht >= 1200 && ht <2500) nht = 7;
+      //else /* ht >= 2500*/             nht = 8;
+      float Zpt = -999;
+      int npt = 0;    
+      // loop through gen parts to identify Z boson
+      int Zidx = -1;
+      for (unsigned int igen = 0; igen < theBigTree.genpart_px->size(); igen++)
+      {
+	bool isZLast   = CheckBit(theBigTree.genpart_flags->at(igen), 13) ; // 13 = isLastCopy
+	bool isZPrompt = CheckBit(theBigTree.genpart_flags->at(igen),  0) ; //  0 = isPrompt
+	if (theBigTree.genpart_pdg->at(igen) == 23 && isZLast && isZPrompt) // Z0 + isLast + isPrompt
+	{
+	  Zidx = igen;
+	}
+      }
+      // if found, Build the genZ TLorentzVector
+      if (Zidx >= 0)
+      {         
+	// build the genZ TLorentzVector
+	TLorentzVector tlvgenZ;
+	tlvgenZ.SetPxPyPzE(theBigTree.genpart_px->at(Zidx), theBigTree.genpart_py->at(Zidx), theBigTree.genpart_pz->at(Zidx), theBigTree.genpart_e->at(Zidx));
+	npt = tlvgenZ.Pt();
+      }
+      if (Zpt >= 50   && Zpt < 100 ) npt = 0;
```

Comment:
Again, is there any particular reason for ignoring the 0To50 sample? Anyway the numbering is wrong because you still have columns 0 to 5, not 4.

Reply:
cf other comment. the numbering will be right if we include back the 0To50 slice

Reply:
I think this was wrong in any case because the table you produce was 5x5x6 so you should consider also the Zpt within 0 and 50 to read the table properly.